ecr/base.go

```diff
@@ -50,36 +51,64 @@ type ecrAPI interface {
 	PutImageWithContext(aws.Context, *ecr.PutImageInput, ...request.Option) (*ecr.PutImageOutput, error)
 }
 
-func (b *ecrBase) getManifest(ctx context.Context) (*ecr.Image, error) {
-	imageIdentifier := b.ecrSpec.ImageID()
-	log.G(ctx).WithField("imageIdentifier", imageIdentifier).Debug("ecr.base.manifest: getting image")
+// getImageByDescriptor retrieves an image from ECR for a given OCI descriptor.
+func (b *ecrBase) getImageByDescriptor(ctx context.Context, desc ocispec.Descriptor) (*ecr.Image, error) {
+	input := ecr.BatchGetImageInput{
+		ImageIds: []*ecr.ImageIdentifier{
+			&ecr.ImageIdentifier{ImageDigest: aws.String(desc.Digest.String())},
+		},
+	}
+	if desc.MediaType != "" {
+		input.AcceptedMediaTypes = []*string{aws.String(desc.MediaType)}
+
+	}
 
-	batchGetImageInput := &ecr.BatchGetImageInput{
-		RegistryId:     aws.String(b.ecrSpec.Registry()),
-		RepositoryName: aws.String(b.ecrSpec.Repository),
-		ImageIds:       []*ecr.ImageIdentifier{imageIdentifier},
+	imgs, err := b.runGetImage(ctx, input)
+	if err != nil {
+		return nil, err
 	}
-	log.G(ctx).WithField("batchGetImageInput", batchGetImageInput).Trace("ecr.base.manifest")
+	return imgs[0], nil
+}
 
-	batchGetImageOutput, err := b.client.BatchGetImageWithContext(ctx, batchGetImageInput)
+func (b *ecrBase) getImage(ctx context.Context) (*ecr.Image, error) {
+	imgs, err := b.runGetImage(ctx, ecr.BatchGetImageInput{
+		ImageIds: []*ecr.ImageIdentifier{b.ecrSpec.ImageID()},
+	})
 	if err != nil {
-		log.G(ctx).WithError(err).Error("ecr.base.manifest: failed to get image")
+		return nil, err
+	}
+	return imgs[0], nil
+}
+
+// runGetImage submits and handles the response for requests of ECR images.
+func (b *ecrBase) runGetImage(ctx context.Context, batchGetImageInput ecr.BatchGetImageInput) ([]*ecr.Image, error) {
```

Comment:
Why return `[]*ecr.Image` when there is validation that the array has only one element?

Can you change this to return `*ecr.Image` so each of the callers doesn't have to do `imgs[0]`?

Reply:
This was trying to be a light wrapper around the API call at the heart of it - returning the list of images wasn't necessary in the end and has been reduced to returning the singular `*ecr.Image`.

Reply:
Should be updated and reflected in the latest push.